File: back-end/pydantec_schemas/teacher.py
from pydantic import BaseModel
from datetime import datetime


class TeacherBase(BaseModel):
    email:str
    nic_number:str

class TeacherCreate(TeacherBase):
    first_name:str
    last_name:str
    address:str
    phone_number:int
class TeacherUpdate(TeacherBase):
    first_name:str
    last_name:str
    address:str
    phone_number:int

class Teacher(TeacherBase):
    teacher_id:int
    created_at:datetime
    updated_at:datetime
    first_name:str
    last_name:str
    address:str
    phone_number:int

    
    class config:
        orm_mode:True
    

File: front-end/src/types/teacher.ts
export type Teacher = {
  first_name: string;
  teacher_id: number;
  last_name: string;
  phone_number: number;
  updated_at: string;
  email: string;
  address: string;
  created_at: string;
  nic_number: string;
};
export interface TeacherForm {
  email: string;
  first_name: string;
  last_name: string;
  address: string;
  phone_number: number;
  nic_number: string;
}

export interface TeacherEditForm {
  email?: string;
  first_name?: string;
  last_name?: string;
  address?: string;
  phone_number?: number;
  nic_number?: string;
}
